src/storage/src/hummock/compaction_group_client.rs

```diff
@@ -49,7 +49,12 @@ impl CompactionGroupClient for CompactionGroupClientImpl {
         &self,
         table_id: StateTableId,
     ) -> HummockResult<CompactionGroupId> {
-        loop {
+        // The loop executes at most twice.
```

Comment:
IMHO, the upper layer (executor) can not handle this Err, this PR just a bug traking ?


Is the panic of the upper layer as expected ？For me , beside using push-based , we need a loop like wait_epoch and some log for traking bug.

Reply:
Could just panic instead of return error.

Reply:
upper layer **may** handle this Err, as it now knows it's working with an in-existent table, mostly likely deleted.